sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5627,6 +5705,12 @@ pi_result piextUSMSharedAlloc(void **ResultPtr, pi_context Context,
       return PI_INVALID_VALUE;
 
     *ResultPtr = It->second.allocate(Size, Alignment);
+    // Keep track of all memory allocations in the platform
+    Plt->MemAllocs.emplace_back(Context, *ResultPtr);
+    // We are going to defer memory release if there are kernels with indirect
+    // access, that is why explicitly retain context to be sure that it is
+    // released after all memory allocations in this context are released.
+    PI_CALL(piContextRetain(Context));
```

Comment:
can't you combine this with the same code above by moving it to before the return?

Reply:
Done, thx.